text/0499-appconfig-constructs.md

```diff
@@ -0,0 +1,422 @@
+    },
+  ],
+});
+```
+
+To import and use a predefined deployment strategy, we can do so as follows.
+
+```ts
+DeploymentStrategy.fromDeploymentStrategyId(this, 'MyPredefinedStrategy', PredefinedDeploymentStrategyId.ALL_AT_ONCE);
+```
+
+# Extension
+
+An extension augments your ability to inject logic or behavior at different points during the AWS AppConfig workflow of creating or deploying a configuration. For example, you can use extensions to perform the following types of tasks (to name a few):
+  * Send a notification to an Amazon Simple Notification Service (Amazon SNS) topic when a configuration profile is deployed.
+  * Scrub the contents of a configuration profile for sensitive data before a deployment starts.
+  * Create or update an Atlassian Jira issue whenever a change is made to a feature flag.
+  * Merge content from a service or data source into your configuration data when you start a deployment.
+  * Back up a configuration to an Amazon Simple Storage Service (Amazon S3) bucket whenever a configuration is deployed.
+
+AWS AppConfig includes the following AWS authored extensions. These extensions can help you integrate the AWS AppConfig workflow with other services. You can use these extensions in the AWS Management Console or by calling extension API actions directly from the AWS CLI, AWS Tools for PowerShell, or the SDK.
+  * Amazon CloudWatch Evidently A/B testing: This extension allows your application to assign variations to user sessions locally instead of by calling the EvaluateFeature operation.
+  * AWS AppConfig deployment events to EventBridge: This extension sends events to the EventBridge default event bus when a configuration is deployed.
+  * AWS AppConfig deployment events to Amazon Simple Notification Service (Amazon SNS): This extension sends messages to an Amazon SNS topic that you specify when a configuration is deployed.
+  * AWS AppConfig deployment events to Amazon Simple Queue Service (Amazon SQS): This extension enqueues messages into your Amazon SQS queue when a configuration is deployed.
+  * Integration extension—Atlassian Jira: This extensions allows AWS AppConfig to create and update issues whenever you make changes to a feature flag.
+
+You can associate these types of tasks with AWS AppConfig applications, environments, and configuration profiles.
+
+For more information about extensions, see [About AWS AppConfig extensions](https://docs.aws.amazon.com/appconfig/latest/userguide/working-with-appconfig-extensions-about.html).
+
+### Example
+```ts
+const extension = new Extension(this, 'MyExtension', {
+  actions: [
+    new Action(this, 'MyAction', {
+      actionPointEvent: <IFunction|IQueue|IBus|ITopic>,  // ex. ActionPointEvent.fromFunction(<IFunction>)
+      actionPoint: ActionPoint.ON_DEPLOYMENT_COMPLETE,
+    }),
+  ],
+
+  // optional
+  name: 'ExtensionName',
+  resource: <IApplciation|IConfigurationProfile|IEnvironment>,   // ex. ExtensionResource.fromApplication(<IApplication>)
```

Comment:
From our docs:
To create an extension, or configure an AWS authored extension, you define the action points that invoke an extension when a specific AWS AppConfig resource is used. For example, you can choose to run the AWS AppConfig deployment events to Amazon SNS extension and receive notifications on an Amazon SNS topic anytime a configuration deployment is started for a specific application. Defining which action points invoke an extension for a specific AWS AppConfig resource is called an extension association. An extension association is a specified relationship between an extension and an AWS AppConfig resource, such as an application or a configuration profile.

So to answer your question, you will choose to associate your extension to the resource(s) that you want to trigger your extension actions. For most cases, this will be associated to an application, but for people with more advanced use cases, they might associate with an environment or configuration profile.